### PR TITLE
#169258646 send notification when trip request approved or rejected

### PR DESCRIPTION
**Pull request by NiyoEric**
#### What does this PR do?
The PR allows a line manager and users to receive a notification when a trip request is approved or rejected.

## **Why**
As travel requests are initiated, notifications should be created within Barefoot Nomad so that managers are aware and can proceed to take approval decisions

**As a requester,** I should be able to see notifications when my travel request has been approved by my manager

#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-backend.git`
- `cd team-odd-bn-backend`
- git checkout -b ft-notification-approve-request-169258646
- `npm install`
- Make sure your `.env` file is updated.
- Drop existing migration `npm run dropTables`
- Update migration `npm run db-migrate`
- Populate the seed data `npm run db-seed-dev`
- `npm run dev-start` check API in development 

```
- use `accept` as query string
- use `reject` as query string
- provide trip request-id  (Ex: 3)

**PUT**  http://localhost:3000/api/v1/trips/3?status=accept
```
- Pass reason in  body
- pass trips info in body
- `npm run test` testing API

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169258646](https://www.pivotaltracker.com/story/show/169258646)

#### Screenshots (if appropriate)

API development in postman
Screen Shot 2019-12-04 at 16 53 32

Send request to browser(Chrome)
Screen Shot 2019-12-05 at 19 03 27

#### Questions:
N/A

### CODE REVIEW

File: src/helpers/socketIo.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    }
+  });
+
+  io.on('connection', (socket) => {
+    eventEmitters.on('notification_message', (data) => {
+      socket.emit('approve_reject', data);
+    });
+  });
+  return io;
+};
```
Comment (hezronkimutai) [inline src/helpers/socketIo.js:21]: Nice work @NiyoEric , I feel like your work will serve as a foundation to the team as we enter the socket.io world

**Review by hezronkimutai — APPROVED** [review]
Well done @NiyoEric.

**Review by victkarangwa — APPROVED** [review]
Well done @NiyoEric 
I like the fact that you built the foundation that we will all be using in the future tasks.

File: src/public/index.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>Home</title>
+</head>
+
+<body>
+  <p>Welcome to Barefoot Nomad</p>
+</body>
+<script src="/socket.io/socket.io.js"></script>
+<script>
+  const socket = io('http://localhost:3000');
```
Comment (ivymwende) [inline src/public/index.html:16]: @NiyoEric Edit this URL to also incorporate production URL on heroku

**Review by ivymwende — COMMENTED** [review]
@NiyoEric Great job! Well done

**Review by julietezekwe — APPROVED** [review]
Great Job Eric

**Pull request merged**